etna/pipeline/pipeline.py

```diff
@@ -54,11 +60,18 @@ def _forecast(self) -> TSDataset:
         if self.ts is None:
             raise ValueError("Something went wrong, ts is None!")
 
-        if isinstance(self.model, DeepBaseModel):
-            future = self.ts.make_future(future_steps=self.model.decoder_length, tail_steps=self.model.encoder_length)
-            predictions = self.model.forecast(ts=future, horizon=self.horizon)
+        if isinstance(self.model, get_args(ContextRequiredModelType)):
+            self.model = cast(ContextRequiredModelType, self.model)
```

Comment:
Why do we need cast here?

Reply:
Because mypy doesn't understant what is `get_args` and don't limit number of classes here. So with `cast` I assure him that we get only these classes in this if close.